addons/initialise-sprite-position/userscript.js

```diff
@@ -19,4 +20,29 @@ export default async function ({ addon }) {
     }
     return oldAddSprite.call(this, stringify ? JSON.stringify(spriteObj) : spriteObj);
   };
+
+  const registerDupPrototype = () => {
+    const targetPrototype = vm.runtime.getTargetForStage().constructor.prototype;
+    const oldDuplicate = targetPrototype.duplicate;
+    targetPrototype.duplicate = function () {
+      return oldDuplicate.call(this).then((newSprite) => {
+        if (!addon.self.disabled) {
+          switch (addon.settings.get("duplicate")) {
+            case "custom":
+              newSprite.setXY(addon.settings.get("x"), addon.settings.get("y"));
+              break;
+            case "keep":
+              newSprite.setXY(this.x, this.y);
+          }
+        }
+        return newSprite;
+      });
+    };
+  };
+
+  if (vm.runtime.getTargetForStage()) {
+    registerDupPrototype();
+  } else {
+    vm.runtime.on(vm.runtime.constructor.PROJECT_LOADED, registerDupPrototype);
```

Comment:
Any reason not to use `vm.runtime.on("PROJECT_LOADED", registerDupPrototype)` like other addons?

Reply:
It could also be `.once` instead of `.on`.

Reply:
Solved my commit